### DISCUSSION TITLE
Auth user id as integer

### DISCUSSION THREAD

**Discussion by MonsterDeveloper** [supabase/supabase#1597] [Questions] [answered]
Hello everyone! I am migrating from MySQL database to Supabase, and I need to migrate my users' data. In my old DB, user id was integer, and I successfully managed to change Supabase auth.users id column type from uuid to int4, as well as storage.objects owner and storage.buckets owner and their constraints on user id.

However, when I open Auth tab in the Supabase Web App, I encounter an error:

![Error at Auth page](https://user-images.githubusercontent.com/16190582/118603895-2e64ec00-b7bd-11eb-81cb-2eb443fa8707.png)

Probably, the app tries to trim the uuid, but as the column is integer now, there is now .trim() method on type number.
So are there any ways to fix or bypass this error and use integer as user ids?

**Comment by steve-chavez** [comment, marked as answer]
Hey @MonsterDeveloper,

I don't recommend modifying any db object on the `auth` or `storage` schemas, they're internal schemas and any change on them might make one of the supabase components fail. We're tightening security and soon we will disallow modifying them.

Instead I'd suggest creating a new table for your mysql users and there have a foreign key to `auth.users`.

**Reply by MonsterDeveloper** [reply]
Oh, that's not cool :( I've changed the other respective fields in other tables and constraints, so it should work with no problems. One error I've encountered so far is the error in Web App, which tries to call .trim() method which doesn't exist on type number.

I could fix that issue, but web app is not open source yet, so its up to Supabase authors to fix this issue. If you're one of them, I kindly ask you to fix that problem, as I can't use Supabase for my project because of it:(

**Reply by kiwicopple** [reply]
Hey @MonsterDeveloper - the Auth system in Supabase uses Netlify's [GoTrue](https://github.com/netlify/gotrue) server. While we would love for the schema to be extremely flexible and adhere to other Auth systems, it would be very difficult without some huge modifications to GoTrue (probably impossible to map our Auth system to every other Auth system).

> I can't use Supabase for my project because of it:(

It is still possible, but it just requires you to map your old users to the "Supabase" system. The steps would be something like this:

- Create a new row in `auth.users` for all of your users. You can store all of your OLD user ID's in the `raw_user_meta_data` field
- Create a `public.users` table if you want to store all your old user info: https://supabase.io/docs/guides/auth#create-a-publicusers-table

There is a bit of work on your side, but our systems are designed to work nicely with Postgres and Row Level Security. It's hard for us to maintain strict security while also offering AAA+ security

**Reply by MonsterDeveloper** [reply]
Thanks, got it. Seems to me like a standard to store ids with uuid and not integer in Postgres.

**Comment by Revadike** [comment]
Would be nice if supabase could offer some flexibility and allow us to choose our preferred user id type/format during initial setup
